pkg/network/kube_secondary_dns_controller.go

```diff
@@ -26,6 +26,13 @@ func renderKubeSecondaryDNS(conf *cnao.NetworkAddonsConfigSpec, manifestDir stri
 	data.Data["NameServerIp"] = conf.KubeSecondaryDNS.NameServerIP
 	data.Data["KubeSecondaryDNSImage"] = os.Getenv("KUBE_SECONDARY_DNS_IMAGE")
 	data.Data["CoreDNSImage"] = os.Getenv("CORE_DNS_IMAGE")
+	if clusterInfo.SCCAvailable {
+		data.Data["RunAsNonRoot"] = "null"
```

Comment:
Passing these null strings looks odd to me and is out of sync with what we do with the other yamls.

Could you please just export the boolean here, maybe

```
data.Data["RunAsNonRoot"] = clusterInfo.SCCAvailable
```

And then us `if` in the template? https://github.com/kubevirt/cluster-network-addons-operator/blob/main/data/linux-bridge/001-rbac.yaml#L1

Reply:
I find this option a bit problematic. 
Something like -
```
{{ if .runAsNonRoot }}
        runAsNonRoot: true
        runAsUser: 1000
{{ end }}
```
has to be injected in the middle of the yaml.
We will need to have a script finding `runAsNonRoot` and `runAsUser` and surrounding it with the condition.
What if for example ksd will add some other param between runAsNonRoot and runAsUser? It will break cnao.

I think setting the `null` values should be safer a cleaner.
https://www.educative.io/answers/how-to-represent-null-values-in-yaml

Reply:
Makes sense. 

Not sure about the way you use null in this PR though. It is quoted and thus a string, not a real null. So it would be evaluated as an invalid value, instead of absent attribute. No?

Reply:
Thanks for the comment. I tested it, it is working.
I deployed `kubeSecondaryDNS` with cnao. The deployment and pods were created and running.
Their yaml doesn't contain both `runAsNonRoot` and `runAsUser`.

Reply:
I see, ok